File: exampleSources/csharp/app1.cs
// C# LINQ + record filtering tasks by status.

using System;
using System.Collections.Generic;
using System.Linq;

public record Task(string Title, Status StatusValue, int Priority)
{
    public enum Status { Open, Done, Blocked }

    public bool IsOpen() => StatusValue == Status.Open;
}

public class Program
{
    public static void Main()
    {
        var backlog = new List<Task>
        {
            new("Write tests", Task.Status.Open, 2),
            new("Fix login bug", Task.Status.Blocked, 5),
            new("Refactor parser", Task.Status.Done, 3),
            new("Ship release", Task.Status.Open, 1),
        };

        var open = backlog
            .Where(t => t.IsOpen())
            .OrderBy(t => t.Priority)
            .Select(t => t.Title);

        foreach (var title in open)
        {
            Console.WriteLine(title);
        }
    }
}

File: exampleSources/java/app1.java
// Java record + stream pipeline filtering tasks by status.

import java.util.List;
import java.util.stream.Collectors;

public record Task(String title, Status status, int priority) {
    public enum Status { OPEN, DONE, BLOCKED }

    public boolean isOpen() {
        return status == Status.OPEN;
    }
}

class Main {
    public static void main(String[] args) {
        List<Task> backlog = List.of(
            new Task("Write tests", Task.Status.OPEN, 2),
            new Task("Fix login bug", Task.Status.BLOCKED, 5),
            new Task("Refactor parser", Task.Status.DONE, 3),
            new Task("Ship release", Task.Status.OPEN, 1)
        );

        List<String> openTitles = backlog.stream()
            .filter(Task::isOpen)
            .sorted((a, b) -> Integer.compare(a.priority(), b.priority()))
            .map(Task::title)
            .collect(Collectors.toList());

        openTitles.forEach(System.out::println);
    }
}